File: frontend/src/constants/repair.ts
import type {RepairStatus} from '../types'; export const REPAIR_STATUS:Record<Uppercase<RepairStatus>,RepairStatus>={PENDING:'pending',ASSIGNED:'assigned',PROCESSING:'processing',DONE:'done',CLOSED:'closed',CANCELLED:'cancelled'}; export const repairStatusText:Record<RepairStatus,string>={pending:'待受理',assigned:'已分派',processing:'处理中',done:'已完成',closed:'已关闭',cancelled:'已取消'};

// 业主仅可在待受理或已分派阶段撤销本人工单；处理中及之后只能走完成/关闭流程。
export const OWNER_CANCELLABLE_STATUSES:RepairStatus[]=['pending','assigned'];
export function isOwnerCancellable(status:RepairStatus):boolean{return OWNER_CANCELLABLE_STATUSES.includes(status)}

File: backend/internal/constants/repair.go
package constants

const (
	RepairStatusPending    = "pending"
	RepairStatusAssigned   = "assigned"
	RepairStatusProcessing = "processing"
	RepairStatusDone       = "done"
	RepairStatusClosed     = "closed"
	RepairStatusCancelled  = "cancelled"
)

var ValidRepairStatuses = map[string]bool{RepairStatusPending: true, RepairStatusAssigned: true, RepairStatusProcessing: true, RepairStatusDone: true, RepairStatusClosed: true, RepairStatusCancelled: true}

// RepairStatusCancellable 业主可自行撤销工单的状态集合：待受理、已分派。
// 处理中及之后的工单只能走完成/关闭流程，不允许业主撤销。
var RepairStatusCancellable = map[string]bool{RepairStatusPending: true, RepairStatusAssigned: true}
